### PR TITLE
Fix range calculation in uniform timestamp generation.

### LINKED ISSUE #2714: [Datagen] Generator generates timestamps out of the specified range
image

### PR DESCRIPTION
**Pull request by gz**
Fixes #2714.

### CODE REVIEW

**Review by mihaibudiu — COMMENTED** [review]
how about supporting SQL timestamps?

File: crates/adapters/src/transport/datagen.rs [hunk 1]
```diff
@@ -2389,6 +2389,40 @@ transport:
...
+            let record = record.unwrap_insert();
+            assert!(
+                record.field >= Timestamp::from_dateTime("2024-10-11T11:04:00Z".parse().unwrap())
+            );
+            assert!(
+                record.field < Timestamp::from_dateTime("2024-10-11T11:05:02Z".parse().unwrap())
+            );
+            assert!(record.field_1 >= Date::new(19963));
+            assert!(record.field_1 < Date::new(19965));
+            assert!(record.field_2 >= Time::new(5000000));
```
Comment (mihaibudiu) [inline crates/adapters/src/transport/datagen.rs:2421]: I have no ideas where these come from

Comment (gz) [reply crates/adapters/src/transport/datagen.rs:2421]: from the datagen config above

**Comment by gz** [comment]
> how about supporting SQL timestamps?

it makes the datagen generator slower .. in a significant way... just did some tests and it's around 80'000 records per second/thread
the reason is that chrono has an efficient parser for everything thats rfc3339 and everything else is slow

**Comment by mihaibudiu** [comment]
your implementation is wrong, the dates have to be parsed exactly once.

**Comment by gz** [comment]
it's fair but thats a bigger change I'll fix it in another issue. I'll need to add a clock strategy anyways for @ryzhyk

**Review by mihaibudiu — APPROVED** [review]

**Pull request merged**